### PR TITLE
Add tracking headers

### PR DESCRIPTION
**Pull request by eccles**
Add DataTrails-User-Agent and DataTrails-Partner-ID headers.

Typical header values when called from datatrails-samples (for example will be:

>{
>    "DataTrails-Partner-ID": "acmecorp",
>    "DataTrails-User-Agent": "datatrails-samples/v0.17.2 pysdk/v0.13.2"
>}

[test_results.txt](https://github.com/user-attachments/files/16099313/test_results.txt)

Fixes [AB#9572](https://dev.azure.com/jitsuin/0629f48c-3979-4bbc-9026-cb06b3dfd0ae/_workitems/edit/9572)

### CODE REVIEW

File: archivist/archivistpublic.py [hunk 1]
```diff
@@ -151,6 +157,16 @@ def max_time(self) -> float:
...
         return self._max_time
 
+    @property
+    def version(self) -> str:
+        """str: Returns version of the archivist package"""
+        return VERSION
+
+    @property
+    def partner_id(self) -> str:
+        """str: Returns partner id of the archivist package"""
```
Comment (JAG-UK) [inline archivist/archivistpublic.py:167]: This looks wrong: surely it's the partner ID the package was initialised with, not some static release property?

File: examples/create_asset.py [hunk 2]
```diff
@@ -97,10 +97,13 @@ def main():
     # Initialize connection to Archivist. max_time is the time to wait for confirmation
     # of an asset or event creation - the default is 300 seconds but one can optionally
     # specify a different value.
+    # The optional partner id field is allocated by Datatrails to partners - partners are then
+    # expected to specify this value when submitting any request to the archivist product.
```
Comment (JAG-UK) [inline examples/create_asset.py:101]:
```suggestion
    # expected to specify this value when submitting any request to the archivist product.
    # Leave blank if you do not have a partner ID
```

**Review by JAG-UK — CHANGES_REQUESTED** [review]

File: archivist/archivistpublic.py (hunk 1 above)
Comment (eccles) [reply archivist/archivistpublic.py]: fixed

File: examples/create_asset.py (hunk 2 above)
Comment (eccles) [reply examples/create_asset.py:101]: fixed

**Review by JAG-UK — APPROVED** [review]

**Pull request merged**